library/core/src/mem/maybe_uninit.rs

```diff
@@ -117,28 +117,42 @@ use crate::slice;
 /// `MaybeUninit<T>` can be used to initialize a large array element-by-element:
 ///
 /// ```
-/// use std::mem::{self, MaybeUninit};
+/// use std::mem::MaybeUninit;
 ///
 /// let data = {
-///     // Create an uninitialized array of `MaybeUninit`. The `assume_init` is
-///     // safe because the type we are claiming to have initialized here is a
-///     // bunch of `MaybeUninit`s, which do not require initialization.
-///     let mut data: [MaybeUninit<Vec<u32>>; 1000] = unsafe {
-///         MaybeUninit::uninit().assume_init()
-///     };
+///     // Create an uninitialized array of `MaybeUninit`.
+///     let mut data = [MaybeUninit::uninit(); 1000];
 ///
-///     // Dropping a `MaybeUninit` does nothing, so if there is a panic during this loop,
-///     // we have a memory leak, but there is no memory safety issue.
 ///     for elem in &mut data[..] {
-///         elem.write(vec![42]);
```

Comment:
Why did you delete the `Vec` example? Is there anything wrong with it?

It's certainly more plausible that a non-copy `u8` wrapper.

Reply:
Yeah, fair enough, restored. I always thought the Vec example was weird but you're right this is no better.